src/StreamableFeesLocker.sol

```diff
@@ -0,0 +1,340 @@
+            position.isUnlocked = true;
+
+            // Transfer the position to the recipient
+            ERC721(address(positionManager)).safeTransferFrom(address(this), position.recipient, tokenId, new bytes(0));
+
+            emit Unlock(tokenId, position.recipient);
+        }
+
+        // Update the position in storage
+        positions[tokenId] = position;
+
+        emit DistributeFees(tokenId, currency0ToDistribute, currency1ToDistribute);
+    }
+
+    /// @notice Releases accrued fees to the caller
+    /// @param tokenId ID of the position to release fees from
+    function releaseFees(
+        uint256 tokenId
+    ) external nonReentrant {
+        // Check if position exists
+        PositionData memory position = positions[tokenId];
+        require(position.startDate != 0, PositionNotFound());
+
+        // Check if sender is a beneficiary
+        bool isBeneficiary = false;
+        for (uint256 i = 0; i < position.beneficiaries.length; i++) {
+            if (position.beneficiaries[i].beneficiary == msg.sender) {
+                isBeneficiary = true;
+                break;
+            }
+        }
+        require(isBeneficiary, NotBeneficiary());
+
+        _releaseFees(tokenId, msg.sender);
+    }
+
+    /// @notice Updates currency balances and calculates distributable amounts
+    /// @param position Current position data
+    /// @param poolKey Pool information
+    /// @return amount0ToDistribute Amount of token0 to distribute
+    /// @return amount1ToDistribute Amount of token1 to distribute
+    function _updateCurrencyBalances(
+        PositionData memory position,
+        PoolKey memory poolKey
+    ) internal returns (uint256 amount0ToDistribute, uint256 amount1ToDistribute) {
+        // Cache currency balances for reentrancy protection
+        uint256 currency0Balance = poolKey.currency0.balanceOfSelf();
+        uint256 currency1Balance = poolKey.currency1.balanceOfSelf();
+
+        // Calculate the amount of fees to distribute
+        amount0ToDistribute = currency0Balance - currencyBalances[poolKey.currency0];
+        amount1ToDistribute = currency1Balance - currencyBalances[poolKey.currency1];
+
+        // Update the global balance
+        currencyBalances[poolKey.currency0] += amount0ToDistribute;
+        currencyBalances[poolKey.currency1] += amount1ToDistribute;
+    }
+
+    /// @notice Distributes fees to a beneficiary
+    /// @param beneficiary Address to distribute fees to
+    /// @param poolKey Pool information
+    /// @param amount0 Amount of token0 to distribute
+    /// @param amount1 Amount of token1 to distribute
+    function _distributeFees(address beneficiary, PoolKey memory poolKey, uint256 amount0, uint256 amount1) internal {
+        beneficiariesClaims[beneficiary][poolKey.currency0] += amount0;
+        beneficiariesClaims[beneficiary][poolKey.currency1] += amount1;
```

Comment:
This is probably pretty obvious, but manual accounting like this means that rebasing tokens will not behave correctly. Not an issue if we use this through the current `Airlock` of course, but maybe worth adding some documentation to indicate this in case of future upgrades allowing for arbitrary tokens or if this contract is used for some other usecase

Reply:
does univ4 support rebasing tokens as the numeraire asset by default? trying to think if even doppler core supports this behaviour or not but otherwise it's definitely worth noting

Reply:
Yeah, realistically I don't think we're ever safe including rebasing tokens in the system, at least I will probably always argue against it